<a href="https://colab.research.google.com/github/Assurance-Project-secure-road/analyse/blob/main/Assurance.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sn

In [75]:
vehicules = pd.read_csv("vehicules-2019.csv", delimiter=";")
vehicules

Num_Acc  id_vehicule num_veh  senc  catv  obs  obsm  choc  manv  \
0       201900000001  138 306 524     B01     2     7    0     2     5    23   
1       201900000001  138 306 525     A01     2    17    1     0     3    11   
2       201900000002  138 306 523     A01     1     7    4     0     1     0   
3       201900000003  138 306 520     A01     1     7    0     2     1     2   
4       201900000003  138 306 521     B01     1     7    1     0     4     2   
...              ...          ...     ...   ...   ...  ...   ...   ...   ...   
100705  201900058837  137 982 135     C01     1     7    0     2     1     2   
100706  201900058838  137 982 132     A01     2     7    3     0     1    21   
100707  201900058839  137 982 131     A01     2    33    0     0     7     1   
100708  201900058840  137 982 129     B01     1    10    0     2     4     0   
100709  201900058840  137 982 130     A01     1    10    0     2     1     0   

        motor  occutc  
0           1     NaN  
1           1     NaN  
2           1     NaN  
3           1     NaN  
4           1     NaN  
...       ...     ...  
100705      1     NaN  
100706      0     NaN  
100707      1     NaN  
100708      2     NaN  
100709      1     NaN  

[100710 rows x 11 columns]

In [76]:
vehicules.describe()

Num_Acc           senc           catv            obs  \
count  1.007100e+05  100710.000000  100710.000000  100710.000000   
mean   2.019000e+11       1.557075      12.632817       0.990646   
std    1.697432e+04       0.833885      12.800755       3.088678   
min    2.019000e+11      -1.000000       0.000000      -1.000000   
25%    2.019000e+11       1.000000       7.000000       0.000000   
50%    2.019000e+11       1.000000       7.000000       0.000000   
75%    2.019000e+11       2.000000      10.000000       0.000000   
max    2.019001e+11       3.000000      99.000000      17.000000   

                obsm           choc           manv          motor      occutc  
count  100710.000000  100710.000000  100710.000000  100710.000000  892.000000  
mean        1.627098       2.926045       7.267401       1.171353    1.818386  
std         1.120068       2.421520       8.077319       1.112887    2.858931  
min        -1.000000      -1.000000      -1.000000      -1.000000    0.000000  
25%         1.000000       1.000000       1.000000       1.000000    1.000000  
50%         2.000000       2.000000       2.000000       1.000000    1.000000  
75%         2.000000       4.000000      15.000000       1.000000    1.000000  
max         9.000000       9.000000      26.000000       6.000000   33.000000

In [77]:
vehicules.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100710 entries, 0 to 100709
Data columns (total 11 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   Num_Acc      100710 non-null  int64  
 1   id_vehicule  100710 non-null  object 
 2   num_veh      100710 non-null  object 
 3   senc         100710 non-null  int64  
 4   catv         100710 non-null  int64  
 5   obs          100710 non-null  int64  
 6   obsm         100710 non-null  int64  
 7   choc         100710 non-null  int64  
 8   manv         100710 non-null  int64  
 9   motor        100710 non-null  int64  
 10  occutc       892 non-null     float64
dtypes: float64(1), int64(8), object(2)
memory usage: 8.5+ MB


# col occutc (Nombre d’occupants dans le transport en commun) 892 enregistrements dans le transport en commun 
# -> le pluspart d'accident arrivent av d'autre type de transport

In [78]:
# remplace valeur -1 par NaN
vehicules = vehicules.replace(-1,np.nan)

In [90]:
# choisir des cols à convertir de type de donnees
list_col = vehicules.columns[3:-1]
list_col

Index(['senc', 'catv', 'obs', 'obsm', 'choc', 'manv', 'motor'], dtype='object')

In [91]:
# convertir le type par Int64
for col in list_col:
    vehicules[col] = vehicules[col].astype("Int64")

# Analyser des donnees

In [79]:
vehicules.isna().value_counts()

Num_Acc  id_vehicule  num_veh  senc   catv   obs    obsm   choc   manv   motor  occutc
False    False        False    False  False  False  False  False  False  False  True      99446
                                                                                False       890
                                                                         True   True        273
                                                    True   False  False  False  True         37
                               True   False  True   True   True   True   True   True         28
                                             False  False  False  False  False  True          4
                               False  False  False  False  True   False  True   True          3
                                             True   True   True   False  True   True          3
                                             False  False  False  True   True   True          2
                                                 

In [57]:
vehicules["catv"].value_counts().sort_index()
#60% nb d'accidents sont des véhicules légères (voitures) - val = 7
# Indeterminable = 0 --> enlever ???? 

0       309
1      5000
2      3716
3       856
7     60787
10     6608
13      405
14      752
15      978
16       39
17      835
20      175
21      250
30     2990
31     1898
32     2399
33     8205
34     1325
35       21
36      150
37      717
38      175
39       32
40      116
41       12
42       35
43      460
50      636
60      141
80      253
99      435
Name: catv, dtype: int64

# enlever les vehicules qui ne sont pas déterminés leur type  (catv=0)

In [58]:
#vehicules = vehicules[vehicules["catv"]!=0]

In [74]:
vehicules.shape

(100401, 11)

In [51]:
#vehicules = vehicules.dropna()

# verifier col occutc correspondant du transport commun av col catv (type de vehicule)

In [60]:
vehicules_commun = vehicules["catv"][(vehicules["catv"] >= 37) & (vehicules["catv"] <= 40)]
vehicules_commun.value_counts()

37    717
38    175
40    116
39     32
Name: catv, dtype: int64

In [73]:
vehicules_avec_occupant = vehicules[vehicules["occutc"]>=0]
vehicules_avec_occupant["catv"].value_counts()

#Les accidents de train et tramway ne sont pas engeristrés les nb occupants

37    717
38    175
Name: catv, dtype: int64

In [26]:
vehicules["obs"].value_counts().sort_index()

0.0     86334
1.0      2314
2.0      1483
3.0      1419
4.0      1376
5.0       163
6.0      1119
7.0       304
8.0      1094
9.0       395
10.0      108
11.0      213
12.0      735
13.0     1744
14.0      798
15.0      537
16.0      399
17.0      134
Name: obs, dtype: int64

In [25]:
vehicules["obsm"].value_counts().sort_index()

0.0    18220
1.0    10380
2.0    70290
4.0       71
5.0       61
6.0      712
9.0      902
Name: obsm, dtype: int64

In [20]:
vehicules["manv"].value_counts().sort_index()

0.0      5947
1.0     40258
2.0     11792
3.0      1166
4.0       481
5.0      1117
6.0       141
7.0       255
8.0        43
9.0      3003
10.0      756
11.0     1201
12.0     1163
13.0     4518
14.0     2029
15.0     8190
16.0     2546
17.0     3598
18.0      550
19.0     2493
20.0      581
21.0     1524
22.0      317
23.0     2763
24.0      393
25.0      238
26.0     3603
Name: manv, dtype: int64

# Type de motorisation du véhicule
- 11.17% de vehicules ne sont pas déterminés leurs types de motorisation 
- 1.45% sont type d'AUTRE 

In [21]:
vehicules["motor"].value_counts().sort_index()

0.0    11215
1.0    80905
2.0      749
3.0     1536
4.0       64
5.0     4464
6.0     1451
Name: motor, dtype: int64

In [65]:
vehicules_motor_inconnu = vehicules["catv"][vehicules["motor"]==0]
vehicules_motor_inconnu.value_counts().sort_index()

1      525
2      400
3      227
7     6829
10     755
13      63
14      44
15      96
16       1
17      56
20      43
21      19
30     344
31     196
32     266
33     646
34     161
35       2
36      13
37     113
38      12
39       7
40      15
41       2
42       3
43      14
50       6
60       7
80       2
99     133
Name: catv, dtype: int64

# Point de choc initial


In [22]:
vehicules["choc"].value_counts().sort_index()

0.0     6038
1.0    36013
2.0    12216
3.0    15028
4.0    10298
5.0     2947
6.0     3612
7.0     5945
8.0     7128
9.0     1440
Name: choc, dtype: int64

In [72]:
vehicules.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 100401 entries, 0 to 100709
Data columns (total 11 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   Num_Acc      100401 non-null  int64  
 1   id_vehicule  100401 non-null  object 
 2   num_veh      100401 non-null  object 
 3   senc         100367 non-null  float64
 4   catv         100401 non-null  int64  
 5   obs          100369 non-null  float64
 6   obsm         100339 non-null  float64
 7   choc         100367 non-null  float64
 8   manv         100368 non-null  float64
 9   motor        100116 non-null  float64
 10  occutc       892 non-null     float64
dtypes: float64(7), int64(2), object(2)
memory usage: 11.2+ MB


In [92]:
vehicules[1150:1200]

Num_Acc  id_vehicule num_veh  senc  catv  obs  obsm  choc  manv  \
1150  201900000693  138 305 267     A01     2     7    2     0     3    13   
1151  201900000694  138 305 265     B01     2     7    0     2     1     1   
1152  201900000694  138 305 266     A01     1     7    0     2     1    13   
1153  201900000695  138 305 263     A01     2    10    0     2     3     1   
1154  201900000695  138 305 264     B01     1    10    0     2     3     1   
1155  201900000696  138 305 262     A01     1     7    4     0     3    13   
1156  201900000697  138 305 260     B01     2     7    0     2     3     1   
1157  201900000697  138 305 261     A01     2     7    0     2     1    26   
1158  201900000698  138 305 259     A01     3     7   10     0     1     0   
1159  201900000699  138 305 257     A01     1    33    0     2     2     2   
1160  201900000699  138 305 258     B01     1    13    0     0     6    11   
1161  201900000700  138 305 256     A01     2     7    2     0     1     1   
1162  201900000701  138 305 254     B01     1     7    0     9     4     2   
1163  201900000701  138 305 255     A01     1     1    0     2     1     2   
1164  201900000702  138 305 253     A01     2     7    0     1     1     1   
1165  201900000703  138 305 251     B01     2     7    0     2     2     1   
1166  201900000703  138 305 252     A01     1     7    0     2     1    15   
1167  201900000704  138 305 249     B01     3     7    0     2     2     1   
1168  201900000704  138 305 250     A01     3     7    8     0     3    16   
1169  201900000705  138 305 247     A01     1     7    0     2     2    17   
1170  201900000705  138 305 248     B01     1    80    0     2     6     1   
1171  201900000706  138 305 245     B01     2     7    0     2     3     0   
1172  201900000706  138 305 246     A01     1     7    0     2     3    13   
1173  201900000707  138 305 244     A01     2     7   13     0     3     0   
1174  201900000708  138 305 242     A01     1     1    0     2     7    12   
1175  201900000708  138 305 243     B01     1    33   15     2     8     1   
1176  201900000709  138 305 241     A01     2     7    0     0     3     0   
1177  201900000710  138 305 240     A01     2     7    0     1     1     1   
1178  201900000711  138 305 239     A01     2     2    0     0     0    26   
1179  201900000712  138 305 238     A01     3     7    0     1     4     4   
1180  201900000713  138 305 237     A01     1    10    8     0     2     1   
1181  201900000714  138 305 236     A01     3    31   12     0     7     1   
1182  201900000715  138 305 235     A01     2     7   13     6     1     1   
1183  201900000716  138 305 233     B01     1     7    0     2     1     1   
1184  201900000716  138 305 234     A01     2     7    0     2     1    15   
1185  201900000717  138 305 231     B01     1     1    0     2     1    15   
1186  201900000717  138 305 232     A01     1     7    0     9     2    17   
1187  201900000718  138 305 230     A01     2     2    0     0     2     1   
1188  201900000719  138 305 228     A01     2     7    0     2     1    12   
1189  201900000719  138 305 229     B01     2     7    5     0     4     1   
1190  201900000720  138 305 226     A01     2     7    0     0     8    10   
1191  201900000720  138 305 227     B01     2    33    0     0     7    17   
1192  201900000721  138 305 221     D01     2     7    0     0     0     1   
1193  201900000721  138 305 222     C01     2     7    0     2     1     1   
1194  201900000721  138 305 223     B01     2     7    0     2     4     1   
1195  201900000721  138 305 224     A01     2     7    0     2     1     1   
1196  201900000722  138 305 219     B01     2    33    0     0     4    15   
1197  201900000722  138 305 220     A01     2    10    0     2     1     2   
1198  201900000723  138 305 217     B01     2    10    0     0     2     0   
1199  201900000723  138 305 218     A01     2     7    0     2     4     1   

      motor  occutc  
1150      